crates/trie/trie/src/node_iter.rs

```diff
@@ -126,6 +128,7 @@ where
     ///
     /// If `metrics` feature is enabled, also updates the metrics.
     fn seek_hashed_entry(&mut self, key: B256) -> Result<Option<(B256, H::Value)>, DatabaseError> {
+        self.last_next_result = None;
```

Comment:
i think there's no need to reset this here

Reply:
Done